spec/lib/tracked_revisions_spec.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_dependency "#{Rails.root}/lib/analytics/course_statistics.rb"
+
+describe 'TrackedRevisions' do
+  let(:course) { create(:course, start: '30-05-2019'.to_date, end: '31-07-2019'.to_date) }
+  let(:user) { create(:user, username: 'Textorus') }
+
+  before do
+    course.students << user
+  end
+
+  it 'fetches the course stats for only tracked articles' do
+    VCR.use_cassette 'course_upload_importer/Textorus' do
+      UpdateCourseStats.new(course)
+    end
+    subject = CourseStatistics.new([course.id]).report_statistics
```

Comment:
I don't think this needs to go through CourseStatistics.

The main things I'd like to see tested are a) the values that end up on the `Course` and `CoursesUsers` records, and b) the revisions (probably just the count) included in the `tracked_revisions` scope. If you test those before and after the untracking and re-updating, that will be a nice test I think.